### PR TITLE
Add roki-glfw & sample

### PR DESCRIPTION
**Pull request by kdaic**
Regarding issue #6, Added a function to use GLFW as an alternative to GLUT. I tried to make the functionality the same as GLUT.
The sample is example/glfw/glfw_solid_test.c. This is just an imitation of example/glut/glut_solid_test.c. However, this sample lacks more examples than the original GLUT one.  
Please confirm.

### CODE REVIEW

**Comment by zhidao** [comment]
I thought it would be better to have common parameters (including a pointer to the default camera) and share them for GLUT and GLFW. Please check src/rkgl_camera.c in the latest version of the main branch. I think you can modify your roki_glfw.c by referring the new roki_glut.c.
Regarding glfw_solid_test.c, rkglBox() is available. See example/shape/box_test.c. Also, why don't you test your rkgl*FuncGLFW() in the program?

**Comment by kdaic** [comment]
roki_glfw was also updated according to the update of rkgl_camera.  
I also used the rkgl_input global variable and updated the callback function.

File: include/roki_gl/roki_glfw.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/* RoKi-GL - Robot Kinetics library: visualization using OpenGL
```
Comment (kdaic) [inline include/roki_gl/roki_glfw.h:1]: hoge

File: src/roki_glfw.c [hunk 2]
```diff
@@ -77,13 +78,13 @@ void rkglKeyFuncGLFW(GLFWwindow *window, int key, int scancode, int action, int
...
 
-void rkglMouseFuncGLFW(GLFWwindow* window, int button, int state, int x, int y)
+void rkglMouseFuncGLFW(GLFWwindow* window, int button, int state, int mods)
 {
   int ctrl_key;
 
   /* get modifier */
   ctrl_key = ( glfwGetKey( window, GLFW_KEY_LEFT_CONTROL ) & GLFW_PRESS ) ? GLFW_KEY_LEFT_CONTROL : 0;
-  rkglMouseStoreInput( button, state, GLFW_PRESS, x, y, ctrl_key );
+  rkglMouseStoreInput( button, state, GLFW_PRESS, rkgl_mouse_x, rkgl_mouse_y, ctrl_key );
```
[the remark this answers is not in the document]
Comment (kdaic) [reply src/roki_glfw.c]: @zhidao  
I Fixed rkglMouseFuncGLFW().   
I define new function ``rkglMouseSotreButtonMod()`` which stores only button( when GLFW_PRESS is enable ) and modifier.   
This function is suggestion for rkgl_input.h/c.

**Comment by zhidao** [comment]
I merged this into the main branch with some modifications.
A major modification is to rename rkglWindowCreateGLFW() to rkglWindowOpenGLFW(), and add another function rkglWindowCreateAndOpenGLFW() in order to avoid confusions.
If you agree with this, I'll close this PR.

**Comment by kdaic** [comment]
@zhidao  
Thanks your merging.  
I agree with the modification that rename rkglWindowOpenGLFW() and add rkglWindowCreateAndOpenGLFW().  
It's OK to close this PR.

**Pull request closed without merging**